LagFreeScreenshots/API/MetadataV2.cs

```diff
@@ -0,0 +1,98 @@
+
+        public override string ToString()
+        {
+            var worldString = "null,Not in any world";
+            if (CurrentInstance.HasValue)
+                worldString = CurrentInstance.Value.WorldId + "," + CurrentInstance.Value.InstanceId;
+
+            var authorString = "?,?";
+            var positionString = "?,?,?";
+            if (CurrentUser.HasValue) {
+                authorString = CurrentUser.Value.Uuid + "," + CurrentUser.Value.Username;
+                var pos = CurrentUser.Value.Transform.position;
+                positionString = $"{FormatNumber(pos.x)},{FormatNumber(pos.y)},{FormatNumber(pos.z)}";
+            }
+
+            return "lfs|cvr|1|author:"
+                + authorString
+                + "|world:" + worldString
+                + "|pos:" + positionString
+                + (ImageRotation != ScreenshotRotation.NoRotation ? "|rq:" + ImageRotation : "")
+                + "|players:" + string.Join(";", PlayerList.Select(PlayerListToString));
+        }
+
+        private static string PlayerListToString(CVRPlayerEntity p)
+        {
+            if (p == null) return "null,0,0,0,null";
+            var pos = p.PlayerObject.transform.position;
+            return $"{p.Uuid},{FormatNumber(pos.x)},{FormatNumber(pos.y)},{FormatNumber(pos.z)},{p.Username}";
+        }
+
+        public static List<CVRPlayerEntity> GetPlayerList(Camera camera)
+        {
+            var localPosition = camera.transform.position;
+
+            // TODO: this won't include current user, is that OK?
+            // TODO: this can be improved a lot (can miss people)
+            var result = MetaPort.Instance.PlayerManager.NetworkPlayers.FindAll(p =>
+            {
+                var avatarRoot = p.PlayerObject.transform;
+                var playerPositionTransform = avatarRoot?.GetComponent<Animator>()?.GetBoneTransform(HumanBodyBones.Head) ?? avatarRoot;
+                var playerPosition = playerPositionTransform.position;
+                Vector3 viewPos = camera.WorldToViewportPoint(playerPosition);
+
+                if (viewPos.z < 2 && Vector3.Distance(localPosition, playerPosition) < 2)
+                {
+                    //User standing right next to photographer, might be visible (approx.)
+                    return true;
+                }
+                else if (viewPos.x > -0.03 && viewPos.x < 1.03 && viewPos.y > -0.03 && viewPos.y < 1.03 && viewPos.z > 2 && viewPos.z < 30)
+                {
+                    //User in viewport, might be obstructed but still...
+                    return true;
+                }
+                return false;
+            }).ToList();
+
+            return result;
+        }
+
+    }
+
+    public struct CurrentPlayerInfo
+    {
+        public string Uuid;
+        public string Username;
+        public Transform Transform;
+    }
+    public struct CurrentInstanceInfo
+    {
+        public string InstanceId;
```

Comment:
The instance name (`MetaPort.Instance.CurrentInstanceName`) could be added to the API as well, would give more clear information instead of only having an id

Reply:
yup also added into the metadata as it will give the world name instead of just the id :3